nav2_controller/src/controller_server.cpp

```diff
@@ -673,20 +681,24 @@ void ControllerServer::updateGlobalPath()
     if (findControllerId(goal->controller_id, current_controller)) {
       current_controller_ = current_controller;
     } else {
-      RCLCPP_INFO(
-        get_logger(), "Terminating action, invalid controller %s requested.",
-        goal->controller_id.c_str());
-      action_server_->terminate_current();
+      std::shared_ptr<Action::Result> result = std::make_shared<Action::Result>();
+      result->error_code = Action::Result::INVALID_CONTROLLER;
+      result->error_msg = "Terminating action, invalid controller " +
+        goal->controller_id + " requested.";
+      RCLCPP_INFO(get_logger(), result->error_msg.c_str());
```

Comment:
I believe `findControllerId` logs this already, so this is a duplicate?

Reply:
I have aimed to keep the logs as they were previously.  I seek to grab them into the result->error_msg where possible so that they can be propogated back to the client.  Because the result->error_msg is a string I use string concatenation on that where possible rather than the printf like services offered by (RCLCP_INFO|WARN|ERROR)

Reply:
I'm saying that you added

```
RCLCPP_INFO(get_logger(), result->error_msg.c_str());
```

But we already log when there's a problem:

```
    if (controllers_.size() == 1 && c_name.empty()) {
      RCLCPP_WARN_ONCE(
        get_logger(), "No controller was specified in action call."
        " Server will use only plugin loaded %s. "
        "This warning will appear once.", controller_ids_concat_.c_str());
      current_controller = controllers_.begin()->first;
    } else {
      RCLCPP_ERROR(
        get_logger(), "FollowPath called with controller name %s, "
        "which does not exist. Available controllers are: %s.",
        c_name.c_str(), controller_ids_concat_.c_str());
      return false;
    }
```

This results in redundant logging. I don't have an issue with `result->error_msg`, just that we're logging that the controller is invalid twice. I don't think the new log is necessary in addition to the populated `error_msg` (for here and in the other 2 `findXYZ()` methods)